libtransmission/crypto-utils.cc

```diff
@@ -106,15 +106,15 @@ int tr_rand_int_weak(int upper_bound)
 {
     TR_ASSERT(upper_bound > 0);
 
-    static bool init = false;
-
-    if (!init)
+    static std::mt19937 random_engine = []()
```

Comment:
I don't think `std::mt19937` is safe to use in multi-threaded environment; would rather declare it `thread_local` instead of `static`. Same for `distribution` below.